tests/metrics/collect_mem_consmd.sh

```diff
@@ -128,38 +134,115 @@ function start_collect()
 # It kills the collectl process.
 function stop_collect()
 {
+	echo "Stop collectl"
 	kill "$COLLECT_PID"
 }
 
-# Get the average about a "csv" file produced
-# by collectl tool.
-function get_average()
+# It launches a instance of psstop tool
+# in order to get metrics about proportional
+# set size.
+function start_psstop()
 {
+	echo "Launching psstop"
+	psstop_monitor=$(cat << EOF
+	#!/bin/bash
+
 	sum=0
 	count=0
 
+	while [ 1 ];do
+		if [ ! -f "$STOP_FILE" ];then
+			data=\$(psstop $PROC | awk '{print \$1}')
+			for i in \$data;do
+				if (( \$i > 0 ));then
+					sum=\$(( \$i + \$sum ))
+					count=\$(( \$count + 1))
+				fi
+			done
+
+			if (( \$count > 0 ));then
+				avg=\$(( \$sum / \$count ))
+				echo "\$avg" >> $PSS_FILE
+			fi
+
+			sleep 0.5
+		else
+			break;
```

Comment:
It's simpler to have the test to break out of the loop immediately after the `while`. That way, you don't need the indent.